Lib/fontTools/feaLib/builder_test.py

```diff
@@ -361,6 +363,92 @@ def test_lookup_inside_feature_aalt(self):
             "Lookup blocks cannot be placed inside 'aalt' features",
             self.build, "feature aalt {lookup L {} L;} aalt;")
 
+    def test_extensions(self):
+        class ast_BaseClass(ast.MarkClass):
+            def asFea(self, indent=""):
+                return ""
+
+        class ast_BaseClassDefinition(ast.MarkClassDefinition):
+            def asFea(self, indent=""):
+                return ""
+
+        class ast_MarkBasePosStatement(ast.MarkBasePosStatement):
+            def asFea(self, indent=""):
+                if isinstance(self.base, ast.MarkClassName) :
+                    res = ""
+                    for bcd in self.base.markClass.definitions :
+                        if res != "" : res += "\n{}".format(indent)
+                        res += "pos base {} {}".format(bcd.glyphs.asFea(), bcd.anchor.asFea())
+                        for m in self.marks :
+                            res += " mark @{}".format(m.name)
+                        res += ";"
+                else :
+                    res = "pos base {}".format(self.base.asFea())
+                    for a, m in self.marks:
+                        res += " {} mark @{}".format(a.asFea(), m.name)
+                    res += ";"
+                return res
+
+        class testAst(object) :
+            MarkBasePosStatement = ast_MarkBasePosStatement
+            def __getattr__(self, name):
```

Comment:
I was wondering if we could turn the `ast` module itself into a class, so that users could subclass it the usual way, instead of using the `__getattr__` trick.
This `ast.AbstractSyntaxTree` class need not be instantiated (unless it holds some state), but could just be a bundle of `@staticmethod` or `@classmethod`, and passed as such to the Parser constructor.